tensorflow/python/ops/array_ops.py

```diff
@@ -3532,7 +3546,76 @@ def _get_paddings_constant(paddings):
     return [_get_paddings_constant(x) for x in paddings]
   else:
     return paddings
+  
+def _wrap_pad(tensor, paddings, name=None):
+  """Helper to create wrap pad option with ops already built.
 
+  See `pad` for more info on how this function works.
+  """
+  # use constant padding to create a tensor with final shape and original values.
+  padded = gen_array_ops.pad_v2(input=tensor, paddings=paddings, name=name)
+  # store the begin and end of original slice area
+  slice_area_to_copy = [
+    _BaseSlice(dim - paddings.numpy()[index][1] - tensor.shape[index], 
```

Comment:
You cannot access `.numpy()` in graph mode (i.e. inside a `tf.function`).  `paddings` will be a symbolic tensor that won't be evaluated until after you build the graph.

Reply:
Similarly, you can't use `tensor.shape`, since it could by dynamic and return `None`.  You need to use `tf.shape(tensor)[index]`

Reply:
This applies everywhere below

Reply:
Updated this.